float64 s
float64 x
float64 y
float64 theta
float64 kappa
float64 left_bound
float64 right_bound
